components/automate-ui/src/app/pages/team-details/team-details.component.ts

```diff
@@ -148,19 +137,15 @@ export class TeamDetailsComponent implements OnInit, OnDestroy {
 
     this.store.select(assignableProjects)
       .subscribe((assignable: ProjectsFilterOption[]) => {
-        assignable.forEach(({ value: id, label: name, type: stringType }) => {
-          const type = <IAMType>stringType;
-          // TODO (tc) Don't actually have the status queried in here but
-          // also don't need it for this page. Maybe the status field
-          // should be optional in the project model?
-          const proj: Project = { id, name, type, status: 'NO_RULES' };
-          // we don't want to override projects that we fetched
-          // that were part of the team already
-          if (!this.projects[proj.id]) {
-            const checked = false;
-            this.projects[proj.id] = { ...proj, checked };
-          }
-        });
+        // compiles a list of all available projects
+        assignable
+          // don't override projects that were already part of the team
+          .filter(p => !this.projects[p.value])
+          .forEach(({ value: id, label: name, type: stringType }) => {
+            const type = <IAMType>stringType;
+            // type and status are unused
+            this.projects[id] = { id, name, type, status: 'NO_RULES', checked: false };
```

Comment:
[nit] The extra const seems superfluous, we could do
```suggestion
            this.projects[id] = { id, name, type: <IAMType>type, status: 'NO_RULES', checked: false };
```
...but it's also not hurting 😉